src/wallet/asyncrpcoperation_shieldcoinbase.h

```diff
@@ -23,29 +23,45 @@
 
 using namespace libzcash;
 
-struct ShieldCoinbaseUTXO {
-    uint256 txid;
-    int vout;
-    CScript scriptPubKey;
-    CAmount amount;
-};
+/**
+When estimating the number of coinbase utxos we can shield in a single transaction:
+1. Joinsplit description is 1802 bytes.
+2. Transaction overhead ~ 100 bytes
+3. Spending a typical P2PKH is >=148 bytes, as defined in CTXIN_SPEND_DUST_SIZE.
+4. Spending a multi-sig P2SH address can vary greatly:
+   https://github.com/bitcoin/bitcoin/blob/c3ad56f4e0b587d8d763af03d743fdfc2d180c9b/src/main.cpp#L517
+   In real-world coinbase utxos, we consider a 3-of-3 multisig, where the size is roughly:
+    (3*(33+1))+3 = 105 byte redeem script
+    105 + 1 + 3*(73+1) = 328 bytes of scriptSig, rounded up to 400 based on testnet experiments.
+*/
+#define CTXIN_SPEND_P2SH_SIZE 400
+
+#define SHIELD_COINBASE_DEFAULT_LIMIT 50
+
+// transaction.h comment: spending taddr output requires CTxIn >= 148 bytes and
+// typical taddr txout is 34 bytes
+#define CTXIN_SPEND_DUST_SIZE   148
```

Comment:
I don't understand why this is named `DUST_SIZE`. However, when making the subsequent ZIP 317 change, we should just use the ZIP 317 constant for transparent input size here.

Reply:
Renamed to `CTXIN_SPEND_P2PKH_SIZE`. Also found that it (and `CTXOUT_REGULAR_SIZE`) had duplicate definitions in rpcwallet.cpp … I’m guessing this PR intended to move them. So I removed those.